src/hb-ot-var-common.hh

```diff
@@ -428,8 +469,23 @@ struct tuple_delta_t
   /* empty for cvar tuples */
   hb_vector_t<float> deltas_y;
 
+  /* compiled data: header and deltas
+   * compiled point data is saved in a hashmap within tuple_variations_t cause
+   * some point sets might be reused by different tuple variations */
+  byte_data_t compiled_tuple_header;
+  byte_data_t compiled_deltas;
```

Comment:
Can you use hb_vector_t<char>'s here? They provide automatic management/ownership of the memory. The vectors could be easily converted to byte_data_t when needed for inclusion in hashmaps.

Reply:
We need to reinterpret_cast between char* and F2DOT14* or HBINT16* when compiling deltas, it's easier if we could just use raw pointers. I'd like to avoid hb_vector_t here if you agree.

Reply:
uh..maybe I can still try to use hb_vector_t, just pre-allocate enough memory and then operate on its raw arrayZ. Would that be better?

Reply:
Yes, that would be better. To handle the casting case you could add a templated helper method which takes hb_vector_t<char>& and returns a hb_array_t<T> (it would cast the arrayZ and create a hb_array_t with it). Then you could easily convert to hb_array_t's of the appropriate type and operate on those instead of the raw pointers.

Reply:
hb_vector_t has operators for hb_array_t if I remember correctly.